dali/pipeline/operator/common.h

```diff
@@ -45,24 +45,19 @@ inline void GetSingleOrRepeatedArg(const OpSpec &spec, vector<T> &result,
 }
 
 template <typename T>
-void GetPerSampleArgument(std::vector<T> &output,
-                          const std::string &argument_name,
-                          const OpSpec &spec,
-                          const ArgumentWorkspace &ws,
-                          int batch_size = -1 /* -1 = get from "batch_size" arg */) {
-  if (batch_size < 0)
-    batch_size = spec.GetArgument<int>("batch_size");
-
+void GetPerSampleArgument(std::vector<T> &output, const std::string &argument_name,
+                          const OpSpec &spec, const ArgumentWorkspace &ws, int batch_size) {
+  DALI_ENFORCE(batch_size > 0,
+               make_string("Invalid batch size. Expected positive, actual: ", batch_size));
```

Comment:
With random application, we must learn to work with empty batches.
I think this should be:
```suggestion
  DALI_ENFORCE(batch_size >= 0);
```
as I can see no way in which the user's code can produce a _negative_ batch size, much less cause an operator to request it.